### PR TITLE
Nessie w/ Iceberg REST / post-release site changes

### LINKED ISSUE #8742: [Catalog] Site + Docs for REST-preview-release

### PR DESCRIPTION
**Pull request by snazy**
Fixes #8742

### CODE REVIEW

File: site/docs/guides/introduction.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+$ select count(*) from t1@mytag join t2@mytag
+.. record 1 ..
+.. record 2 ..
+.. only 2 records ..
+```
+
+## Data and Metadata
+
+Nessie does not make copies of your underlying data. Instead, it works to version
```
Comment (adutra) [inline site/docs/guides/introduction.md:70]:
```suggestion
Nessie does not make copies of your underlying data. Instead, it works by versioning
```

File: site/docs/guides/introduction.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+...
+$ insert 2 records into t1
+...
+$ create t2
+...
+$ insert 2 records into t2
+...
+```
+
+A user can then use the Nessie CLI to view the history of the main branch. You'll see
```
Comment (adutra) [inline site/docs/guides/introduction.md:36]: Is this referring to Pynessie? Why not use the new CLI for these examples?

File: site/docs/guides/introduction.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+table metadata management.
+
+## Technology 
+Nessie can be [deployed in multiple ways](../downloads/index.md) and is composed primarily of the Nessie service, 
+which exposes a set of [REST APIs](../develop/rest.md) and a simple browser UI. This service works with multiple
+libraries to expose Nessie's version control capabilities to common data management technologies.
+
+Nessie was built as a Cloud native technology and is designed to be highly scalable, 
+[performant](../develop/kernel.md) and resilient. It is built 
+on Java and leveraging [Quarkus](https://quarkus.io/).
```
Comment (adutra) [inline site/docs/guides/introduction.md:92]:
```suggestion
on Java and leverages [Quarkus](https://quarkus.io/).
```

File: site/docs/guides/introduction.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+This makes Nessie work very well in Docker and FaaS environments. 
+Nessie has a pluggable storage backend and comes pre-packaged with support for Amazon's DynamoDB,
+Google BigTable, Apache Cassandra, ScyllaDB, PostgreSQL (incl. Cockroach) and local storage
```
Comment (adutra) [inline site/docs/guides/introduction.md:95]:
```suggestion
Google BigTable, Apache Cassandra, ScyllaDB, PostgreSQL (incl. Cockroach), MariaDB, MySQL and local storage
```

File: site/docs/guides/try-nessie.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Getting Started with Nessie and Iceberg REST
+
+Nessie supports Apache Iceberg REST! Some of the features are
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:3]:
```suggestion
"Nessie Catalog" is the new Nessie-backed implementation of the Apache Iceberg REST Catalog! Some of its features are:
```

File: site/docs/guides/try-nessie.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+* All clients that "speak" Iceberg REST can work with Nessie, whether it's running under Java, Scala, Python or Rust.
+* Nessie Catalog S3 and compatible object stores like [MinIO](https://min.io/), supporting both request signing and
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:6]:
```suggestion
* Nessie Catalog supports Amazon S3 and compatible object stores like [MinIO](https://min.io/), including request signing and
```

File: site/docs/guides/try-nessie.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# Getting Started with Nessie and Iceberg REST
+
+Nessie supports Apache Iceberg REST! Some of the features are
+
+* All clients that "speak" Iceberg REST can work with Nessie, whether it's running under Java, Scala, Python or Rust.
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:5]:
```suggestion
* All clients that "speak" Iceberg REST can work with Nessie Catalog, whether it's running under Java, Scala, Python or Rust.
```

File: site/docs/guides/try-nessie.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+* Nessie Catalog S3 and compatible object stores like [MinIO](https://min.io/), supporting both request signing and
+  session tokens.
+* (Currently experimental) support for Google Cloud Storage and ADLS Gen2.
+* Plus all the existing Nessie features.
+
+To just give Nessie Catalog, so Nessie with Iceberg REST, a try we prepared a Docker/Podman compose file for you.
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:11]:
```suggestion
For those wanting to give Nessie Catalog a try, we prepared a Docker/Podman compose file for you.
```

File: site/docs/guides/try-nessie.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+Running Nessie on your laptop and access it using Spark SQL and Iceberg REST is not difficult. The following starts
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:12]:
```suggestion
Running Nessie on your laptop and accessing it using Spark SQL and Iceberg REST is not difficult. The following starts
```

File: site/docs/guides/try-nessie.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  --packages "org.projectnessie.nessie-integrations:nessie-spark-extensions-3.5_2.12:{{ versions.nessie }},org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.5.2" \
+  --conf spark.sql.extensions=org.projectnessie.spark.extensions.NessieSparkSessionExtensions,org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions \
+  --conf spark.sql.catalogImplementation=in-memory \
+  --conf spark.sql.catalog.nessie.uri=http://127.0.0.1:19120/iceberg/main/ \
+  --conf spark.sql.catalog.nessie.type=rest \
+  --conf spark.sql.catalog.nessie=org.apache.iceberg.spark.SparkCatalog
+```
+
+!!! note
+    With Nessie all necessary configuration about the object store, in the above example it is S3 via MinIO, is
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:35]:
```suggestion
    With Nessie, all necessary configuration about the object store (in the above example, it is S3 via MinIO), is
```

File: site/docs/guides/try-nessie.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    pushed from Nessie to the Iceberg client used by Spark. Requests from Spark/Iceberg to the object store are
+    "secured" via Nessie, in the above example using S3 request signing. This means, that you do not configure
+    object store credentials on your Iceberg REST clients.
+
+And that's basically all to connect to Nessie using Iceberg REST and try it out _locally_ - from there it just works.
+
+The above docker/podman-compose configuration does a bunch of things for the convenience for a demo/try-out.
```
Comment (adutra) [inline site/docs/guides/try-nessie.md:42]:
```suggestion
The above docker/podman-compose configuration does a bunch of things for the convenience of a demo/try-out.
```

File: site/docs/guides/introduction.md (hunk 3 above)
Comment (snazy) [reply site/docs/guides/introduction.md:92]: Written by a native english speaker :P

File: site/docs/guides/introduction.md (hunk 2 above)
Comment (snazy) [reply site/docs/guides/introduction.md:36]: It's not new (moved from index.md) - let's defer this to another PR.

File: site/docs/guides/try-nessie.md (hunk 5 above)
Comment (snazy) [reply site/docs/guides/try-nessie.md:3]: Why introduce a new "Nessie _Catalog_" to users?

File: site/docs/guides/introduction.md (hunk 3 above)
Comment (dimas-b) [reply site/docs/guides/introduction.md:92]: my 2 cents is for `leverages`

**Review by dimas-b — DISMISSED** [review]

File: site/docs/guides/iceberg-rest.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+  returns the Iceberg view version corresponding to the Nessie commit.
+
+## Nessie CLI
+
+The [Nessie CLI](../nessie-latest/cli.md) has been enhanced with basic support for Iceberg REST when
+used with Nessie. It will transparently connect with Iceberg REST as well. You can also use the
+Iceberg REST base URI instead of the Nessie REST base URI.
+
+!!! tip
+    If you use `CONNECT TO http://127.0.0.1:19120/api/v2 USING "nessie.authentication.type" = BEARER`,
```
[the remark this answers is not in the document]
Comment (snazy) [reply site/docs/guides/iceberg-rest.md:99]: Better not in the context, it's mentioned around line 73 though

[the remark this answers is not in the document]
Comment (dimas-b) [reply site/docs/guides/iceberg-rest.md:100]: Ah, ok... I (apparently erroneously) assumed that the CLI would be able to execute Device Code Flow via the Nessie Client.

**Review by dimas-b — DISMISSED** [review]

**Review by dimas-b — APPROVED** [review]

**Pull request merged**